apps/nextjs-app-router/src/makeswift/components/css-theme/index.tsx

```diff
@@ -0,0 +1,30 @@
+import { MakeswiftComponent } from '@makeswift/runtime/next'
+import { type ComponentPropsWithoutRef } from 'react'
+
+import { getComponentSnapshot } from '@/makeswift/client'
+
+import { CssTheme as CssThemeClient, PropsContextProvider } from './client'
+import { COMPONENT_TYPE } from './register'
+
+type Props = ComponentPropsWithoutRef<typeof CssThemeClient> & {
+  snapshotId?: string
+  label?: string
+}
+
+export const CssTheme = async ({
```

Comment:
The naming here was hard for me to understand. Maybe this could be `HeadRootComponent` or something?

My understanding is this is the slot in our `<head/>` and we drop the other `CssComponent` into this slot and it renders the CSS.

Reply:
> The naming here was hard for me to understand. Maybe this could be HeadRootComponent or something?

The pattern used here (server component wrapping a client component embedded in the snapshot) is what we settled on in Catalyst for defining all "embedded" components in general. This particular one is named for what it does (defines a CSS theme) rather where you'd use it, which I think what components should be named. Does it makes sense?

> My understanding is this is the slot in our <head/> and we drop the other CssComponent into this slot and it renders the CSS.

This a server component that is tied to a specific component snapshot, `site-theme`, and that component snapshot, once edited, will store the element tree for our client component registered under `COMPONENT_TYPE`.